### PR TITLE
Terminals

### PR DESCRIPTION
**Pull request by jdetle**
The tests currently aren't working and I'd love an extra pair of eyes to help me figure out why, thanks.

The current error is saying that `list creates an AjaxObservable for listing available terminals:
     TypeError: Cannot read property 'url' of undefined`

### CODE REVIEW

File: test/terminals-spec.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+const serverConfig = {
+  endpoint: 'http://localhost:8888',
+  crossDomain: true,
+};
+
+describe('list', () => {
+  it('creates an AjaxObservable for listing available terminals', () => {
+    const list$ = terminals.list(serverConfig);
+    const request = list$.response;
```
Comment (rgbkrk) [inline test/terminals-spec.js]: These are all failing because you need to use `request`, not `response`.

```
const request = list$.request;
```

Comment (jdetle) [inline test/terminals-spec.js]: Hah _thank you_

**Comment by jdetle** [comment]
@rgbkrk if you could take care of the websockets stuff I'm not entirely sure how to handle it.

File: src/terminals.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+/**
+ * [webSocket description]
+ * @param  {[type]} basePath [description]
+ * @param  {[type]} name     [description]
+ * @return {[type]}          [description]
+ */
+export function webSocket(basePath, name) {
+  const uri = `ws://${basePath}/terminals/webSocket/${name}`;
```
Comment (rgbkrk) [inline src/terminals.js]: This should be lowercase `/websocket/` in the url.

Also, you'll need to do the same handling as we do in kernels for [creating the websocket URL](https://github.com/nteract/rx-jupyter/blob/master/src/kernels.js#L91-L98).

I'd probably only set this up as a helper for `formWebSocketURL` and not create an Rx WebSocket Subject.

**Comment by jdetle** [comment]
@rgbkrk ready for merge?

File: src/terminals.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+}
+/**
+ * Return an rxjs websocket for a given id.
+ * @param {Object} serverConfig  - The server configuration.
+ * @param  {string} id - ID of the terminal to have a new websocket created.
+ * @return {webSocket} Websocket for the terminal to with specified id.
+ */
+export function connect(serverConfig, id) {
+  return webSocket(formWebSocketURL(serverConfig, id));
+}
```
Comment (rgbkrk) [inline src/terminals.js]: Based on [how these get used by xterm.js](http://xtermjs.org/docs/guides/terminado/), we shouldn't create our own websocket subject. Only export and test `formWebSocketURL`.

Comment (jdetle) [inline src/terminals.js]: Missed this, will fix.

**Comment by jdetle** [comment]
👍 thanks for the review!

**Pull request merged**